src/js/user.js

```diff
@@ -63,25 +87,40 @@ function UserManager(auth, server) {
         },
 
         createRepo: function (user, reponame) {
-            return api.searchRepos("_", user.id, 1000).then(_.flatten).then(function (repos) {
-                return _.find(repos, {full_name: user.username + '/' + reponame});
-            }).then(function (repo) {
-                return repo ? repo : api.createRepo({
-                    name: reponame,
-                    description: 'btt-writer-desktop: ' + reponame,
-                    private: false
-                }, user);
-            });
+            let query = "_";
+
+            return fetchRepoRecursively(user.id, query, MAX_PAGE_SIZE)
+                .then(function (repos) {
+                    return _.find(repos, {full_name: user.username + '/' + reponame});
+                })
+                .then(function (repo) {
+                    return repo ? repo : api.createRepo({
+                        name: reponame,
+                        description: 'btt-writer-desktop: ' + reponame,
+                        private: false
+                    }, user);
+                });
         },
 
         retrieveRepos: function (u, q) {
             u = u === '*' ? '' : (u || '');
             q = q === '*' ? '_' : (q || '_');
 
-            var limit = 100;
+            let defaultLimit = 10;
```

Comment:
The default limit of WACS is 50 results per page -- I recommend we ask for 50 results instead of 10.

Reply:
the `defaultLimit` was used to search for all repos (uid = 0 is search all) that matches the query `q`.
The use case of this is particularly for searching by project query, so I think top 10 matching results is a good number.
More significantly, line 123 where it searches for matching user `u`, then the result of this would be multiply with the repos found of each user. Eventually, the number would be quite large, which I accidentally left it = 100 and it takes minutes to search.